adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -272,12 +284,41 @@ def _real_do(self) -> PerceptualRepresentation[DevelopmentalPrimitivePerceptionF
             for relation in self._situation.after_action_relations
         ]
 
+        # Add ground for always perception if needed
+        if self._include_ground:
+            self._relation_perceptions.extend(
+                self._perceive_ground_relations(
+                    relations=chain(
+                        self._relation_perceptions,
+                        explicit_before_relations,
+                        _action_perception.before_relations,
+                        explicit_after_relations,
+                        _action_perception.after_relations,
+                    )
+                )
+            )
+            before_ground = self._perceive_ground_relations(
```

Comment:
Add a comment indicating why we need to handle `before_ground` and `after_ground` separately for a future reader.